### DISCUSSION TITLE
Subcircuit symbol contains overlay of "schematic" and symbol

### DISCUSSION THREAD

**Discussion by tomhajjar** [ra3xdh/qucs_s#617] [Q&A] [answered]
While making a dual-gate MOSFET subcircuit, something occurred I don't remember seeing before. If I choose Spice netlist versus SpiceLibComp, the model "schematic" and dual-gate MOSFET symbol are overlaid when placed on schematic. The OpAmp library was made using Spice netlist and never had this problem.

![2024-02-27_200501](https://github.com/ra3xdh/qucs_s/assets/102838604/9e8b4873-16a9-48de-8db1-1b06b7e5e003)

![2024-02-27_204026](https://github.com/ra3xdh/qucs_s/assets/102838604/5019a0e1-cdd2-45bc-a8ce-238304a67ff5)

**Comment by ra3xdh** [comment, marked as answer]
@tomhajjar , I have just tried to recreate the device from scratch using SPICE file and I cannot reproduce the problem. Everything works as expected. I have also inspected my projects containing subcircuits made from SPICE file. Everything works as expected too.  I suspect the schematic file became broken by unknown reason. I sucpect this may be related to the recent schematic class refactoring. 

Also please keep in mind that Qucs-S contains BF998 model since v0.0.23 release. You don't need to recreate it. I don't consider the update of this library. Other dual gate MOSFETs like BF961 could be added, but I didn't find the SPICE models for them.

**Comment by ra3xdh** [comment]
@tomhajjar You have incorrect port number `Num` property. Set the number here instead of `G2` to fix the issue.  This property must be set only to integer numbers. Otherwise you will get the broken schematic. Unfortunately there is no possibility to add a validator for this property. I don't recommend changing automatically assigned port numbers. But port names could be changed. 
![image](https://github.com/ra3xdh/qucs_s/assets/4920080/0ba720b1-4b7c-4132-9982-584c25819d79)

**Comment by tomhajjar** [comment]
Thanks for the catch. I was porting an LTspice project by Gunthard Kraus over to Qucs-S and forgot BF998 was native. I also wanted to add a dual-gate device to schematic_symbols.sch.

The LTspice example deletes the C=/L= entries in BF998.cir. ngspice doesn't say in the manual this format is acceptable but it must be since I confirmed the S-parameters are the same for the library model and the subcircuit I made.

![2024-02-28_070846](https://github.com/ra3xdh/qucs_s/assets/102838604/23c8f39a-4508-4d78-82ac-3cb8603a3664)

**Comment by tomhajjar** [comment]
Here is Bordodynov's dual gate mosfet library.

[Dual-gate MOSFET.zip](https://github.com/ra3xdh/qucs_s/files/14435147/Dual-gate.MOSFET.zip)

**Comment by tomhajjar** [comment]
I found the s-parameters for the BF998 and they compare favorably with the Spice model. 

I converted a LTspice 137MHz LNA project by the late Gunthard Kraus over to Qucs-S. S11 looks sketchy but it matches the original data where s11 -> NFopt.

https://www.gunthard-kraus.de/LTSwitcherCAD/SwitcherCAD-Tutorial_English/pdf-File/RF_English_01_final.pdf

![2024-02-28_160918](https://github.com/ra3xdh/qucs_s/assets/102838604/3b9fcade-f833-47ab-97de-d823ac47e494)

![2024-02-28_160841](https://github.com/ra3xdh/qucs_s/assets/102838604/16f26204-859e-48a0-bb2b-eb418cb08a38)

[BF998_MOSFET_prj.zip](https://github.com/ra3xdh/qucs_s/files/14439958/BF998_MOSFET_prj.zip)
